# Hamoye Python for Machine Learning

In [3]:
import pandas as pd
import numpy as np

In [4]:
df = pd.read_csv('FoodBalanceSheets_E_Africa_NOFLAG.csv', encoding = 'latin-1')
df.describe()

Area Code     Item Code  Element Code          Y2014          Y2015  \
count  60943.000000  60943.000000  60943.000000   59354.000000   59395.000000   
mean     134.265576   2687.176706   3814.856456     134.196282     135.235966   
std       72.605709    146.055739   2212.007033    1567.663696    1603.403984   
min        4.000000   2501.000000    511.000000   -1796.000000   -3161.000000   
25%       74.000000   2562.000000    684.000000       0.000000       0.000000   
50%      136.000000   2630.000000   5142.000000       0.090000       0.080000   
75%      195.000000   2775.000000   5511.000000       8.340000       8.460000   
max      276.000000   2961.000000   5911.000000  176405.000000  181137.000000   

               Y2016          Y2017          Y2018  
count   59408.000000   59437.000000   59507.000000  
mean      136.555222     140.917765     143.758381  
std      1640.007194    1671.862359    1710.782658  
min     -3225.000000   -1582.000000   -3396.000000  
25%         0.000000       0.000000       0.000000  
50%         0.080000       0.100000       0.070000  
75%         8.430000       9.000000       9.000000  
max    185960.000000  190873.000000  195875.000000

In [6]:
df.head()

Area Code     Area  Item Code         Item  Element Code  \
0          4  Algeria       2501   Population           511   
1          4  Algeria       2501   Population          5301   
2          4  Algeria       2901  Grand Total           664   
3          4  Algeria       2901  Grand Total           674   
4          4  Algeria       2901  Grand Total           684   

                                  Element             Unit     Y2014  \
0           Total Population - Both sexes     1000 persons  38924.00   
1                Domestic supply quantity      1000 tonnes      0.00   
2           Food supply (kcal/capita/day)  kcal/capita/day   3377.00   
3  Protein supply quantity (g/capita/day)     g/capita/day     94.90   
4      Fat supply quantity (g/capita/day)     g/capita/day     80.06   

      Y2015     Y2016     Y2017     Y2018  
0  39728.00  40551.00  41389.00  42228.00  
1      0.00      0.00      0.00      0.00  
2   3379.00   3372.00   3341.00   3322.00  
3     94.35     94.72     92.82     91.83  
4     79.36     77.40     80.19     77.28

In [7]:
df.isna().sum()

Area Code          0
Area               0
Item Code          0
Item               0
Element Code       0
Element            0
Unit               0
Y2014           1589
Y2015           1548
Y2016           1535
Y2017           1506
Y2018           1436
dtype: int64

Number and percentage of missing data in 2014 to 3dp

In [32]:
df['Y2014'].isnull().sum()

1589

In [18]:
round((df['Y2014'].isnull().sum()*100/len(df)),3)

2.607

## Perform groupby on element. Total sum of processing in 2017

In [19]:
df.groupby('Element')['Y2017'].sum()

Element
Domestic supply quantity                  2088198.10
Export Quantity                            182338.80
Fat supply quantity (g/capita/day)          10253.84
Feed                                       223705.68
Food                                      1258888.28
Food supply (kcal/capita/day)              454681.00
Food supply quantity (kg/capita/yr)         48690.04
Import Quantity                            294559.09
Losses                                     160614.00
Other uses (non-food)                       91645.97
Processing                                 292836.00
Production                                2030056.89
Protein supply quantity (g/capita/day)      11842.45
Residuals                                   35500.00
Seed                                        24870.14
Stock Variation                             54316.91
Total Population - Both sexes             1112641.00
Tourist consumption                            91.00
Name: Y2017, dtype: float64

## Highest sum area in 2017

In [20]:
df.groupby('Area')['Y2017'].sum()

Area
Algeria                         325644.27
Angola                          229159.57
Benin                           124771.22
Botswana                         22101.30
Burkina Faso                    101855.07
Cabo Verde                       14650.74
Cameroon                        232030.43
Central African Republic         29937.00
Chad                             71594.68
Comoros                             59.84
Congo                            41181.68
Côte d'Ivoire                   224599.01
Djibouti                         22729.91
Egypt                           866379.92
Eswatini                         54343.33
Ethiopia                        448683.76
Ethiopia PDR                         0.00
Gabon                            27979.64
Gambia                           23154.18
Ghana                           337599.06
Guinea                           98138.87
Guinea-Bissau                    19102.77
Kenya                           264660.66
Lesotho                      

In [21]:
df.groupby('Area')['Y2017'].sum().max()

1483268.23

## Area with 7th lowest sum in 2017

In [22]:
sorted(df.groupby('Area')['Y2017'].sum())

[0.0,
 0.0,
 59.839999999999996,
 442.34000000000003,
 12662.630000000001,
 14650.74,
 19102.77,
 21267.96,
 22101.3,
 22729.91,
 23154.18,
 27979.64,
 29342.2,
 29874.89,
 29937.0,
 41181.68,
 49841.88,
 51114.83,
 54343.33,
 55311.33,
 71594.68,
 73663.69,
 75919.34,
 95681.15,
 98138.87,
 101855.06999999999,
 103223.77,
 124167.2,
 124771.22,
 126707.58,
 131197.73,
 149928.33,
 156665.46,
 161407.98,
 181098.71,
 213950.38,
 224599.01,
 229159.57,
 232030.43,
 239931.92,
 264660.66,
 322616.85,
 325644.27,
 337599.06,
 388495.36,
 448683.76,
 517590.54,
 866379.92,
 1483268.23]

7th lowest value belongs 19102.77

## Total protein supply in madagascar in 2015

In [24]:
df2 = df.groupby(['Area', 'Element'], as_index=False)['Y2015'].sum()
df2.iloc[410:430, :]

Area                                 Element     Y2015
410     Liberia           Total Population - Both sexes   4472.00
411  Madagascar                Domestic supply quantity  31214.98
412  Madagascar                         Export Quantity    494.75
413  Madagascar      Fat supply quantity (g/capita/day)     91.85
414  Madagascar                                    Feed   2070.22
415  Madagascar                                    Food  21120.65
416  Madagascar           Food supply (kcal/capita/day)   7685.00
417  Madagascar     Food supply quantity (kg/capita/yr)    871.59
418  Madagascar                         Import Quantity   1721.80
419  Madagascar                                  Losses   2176.00
420  Madagascar                   Other uses (non-food)   3957.12
421  Madagascar                              Processing   1803.00
422  Madagascar                              Production  29482.89
423  Madagascar  Protein supply quantity (g/capita/day)    173.05
424  Madagascar                               Residuals   -616.00
425  Madagascar                                    Seed    699.00
426  Madagascar                         Stock Variation   -505.00
427  Madagascar           Total Population - Both sexes  24234.00
428      Malawi                Domestic supply quantity  43271.06
429      Malawi                         Export Quantity    583.02

# Least correlation with element code

In [26]:
correlation = df.corr()
correlation

Area Code  Item Code  Element Code     Y2014     Y2015  \
Area Code      1.000000  -0.005159     -0.000209  0.006164  0.005472   
Item Code     -0.005159   1.000000     -0.024683  0.021722  0.020857   
Element Code  -0.000209  -0.024683      1.000000  0.024457  0.023889   
Y2014          0.006164   0.021722      0.024457  1.000000  0.994647   
Y2015          0.005472   0.020857      0.023889  0.994647  1.000000   
Y2016          0.005247   0.020109      0.023444  0.996081  0.995739   
Y2017          0.005006   0.021494      0.024254  0.995230  0.988048   
Y2018          0.005665   0.021314      0.024279  0.994872  0.988208   

                 Y2016     Y2017     Y2018  
Area Code     0.005247  0.005006  0.005665  
Item Code     0.020109  0.021494  0.021314  
Element Code  0.023444  0.024254  0.024279  
Y2014         0.996081  0.995230  0.994872  
Y2015         0.995739  0.988048  0.988208  
Y2016         1.000000  0.992785  0.992757  
Y2017         0.992785  1.000000  0.998103  
Y2018         0.992757  0.998103  1.000000

In [28]:
correlation.iloc[3:,2]

Y2014    0.024457
Y2015    0.023889
Y2016    0.023444
Y2017    0.024254
Y2018    0.024279
Name: Element Code, dtype: float64

# unique countries 

In [29]:
df.groupby(['Area Code'], as_index=False)['Area'].count()

Area Code  Area
0           4  1313
1           7  1424
2          20  1378
3          32  1460
4          35  1260
5          37  1344
6          39  1343
7          45    45
8          46  1413
9          53  1285
10         59  1488
11         62    39
12         72  1095
13         74  1245
14         75  1168
15         81  1354
16         90  1410
17        107  1467
18        114  1560
19        122  1150
20        123  1206
21        129  1439
22        130  1447
23        133  1293
24        136  1195
25        137  1378
26        143  1393
27        144  1427
28        147  1283
29        158  1403
30        159  1474
31        175  1128
32        181  1393
33        184  1426
34        193  1177
35        195  1462
36        196    68
37        197  1242
38        202  1399
39        206    53
40        209  1284
41        215  1419
42        217  1294
43        222  1338
44        226  1458
45        233  1409
46        238  1447
47        251  1451
48        276  1316

# Mean and SD of 2017 to 2 dp

In [30]:
round(df['Y2017'].mean(),2)

140.92

In [31]:
round(df['Y2017'].std(axis=0),2)

1671.86

# sum of wine in produced in 2015 and 2018

In [10]:
total_wine=df.groupby('Item')[['Y2015','Y2018']].sum()
print(total_wine.tail())

                        Y2015      Y2018
Item                                    
Vegetables, Other   158104.08  163987.21
Vegetal Products    107064.17  107775.39
Wheat and products  234710.51  242645.19
Wine                  4251.81    4039.32
Yams                203151.78  221272.09
